app/components/candidate_interface/gcse_qualification_review_component.rb

```diff
@@ -33,6 +33,13 @@ def show_missing_banner?
       end
     end
 
+    def show_values_missing_banner?
+      if @submitting_application
+        gcse_completed = "#{@subject}_gcse_completed"
+        (!@application_form.send(gcse_completed) || @application_form.send("#{@subject}_gcse").incomplete_gcse_information?) && @editable && !@application_qualification.missing_qualification?
```

Comment:
Looking at this with fresh eyes. Can we pull this into a separate method so it's a bit clearer what is going on?

We can put this block

`(!@application_form.send(gcse_completed) || @application_form.send("#{@subject}_gcse").incomplete_gcse_information?)`

into a private method called `section_or_gcse_is_incomplete?` or something like that

Reply:
On second thought we can also add this here too `!@application_qualification.missing_qualification?` as the method name covers the behaviour.

So 

```
private

def section_or_gcse_is_incomplete?
  (!@application_form.send(gcse_completed) || 
    @application_form.send("#{@subject}_gcse").incomplete_gcse_information?) && 
    !@application_qualification.missing_qualification?
end 

```